tests/Test/Xrefcheck/URIParsingSpec.hs

```diff
@@ -0,0 +1,47 @@
+{- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.URIParsingSpec where
+
+import Universum
+
+import Test.Hspec (Spec, describe, it)
+import Test.HUnit (assertBool)
+import URI.ByteString qualified as URIBS
+
+parse :: Text -> Either URIBS.URIParseError (URIBS.URI)
+parse link = URIBS.parseURI URIBS.laxURIParserOptions (encodeUtf8 link)
+
+spec :: Spec
+spec = do
+  describe "URI parsing should be successful" $ do
+    it "Without the special characters in the query strings" $
+      forM_ withoutSpecChars $ \link -> do
+        assertBool "Failed to parse the link" (isRight $ parse link)
```

Comment:
Well, looks like these tests are testing the local `parse` function... they're not testing xrefcheck.

If the code in `Verify` changes, this test will continue passing. We won't catch any possible bugs.

Reply:
I suggest extracting those two lines in `checkExternalResource` into a top-level function, e.g. `parseURL :: Text -> ExceptT VerifyError IO URI`, and then export that function so it can be tested here.